pandas/tests/frame/test_join.py

```diff
@@ -165,3 +170,11 @@ def test_join_period_index(frame_with_period_index):
         index=frame_with_period_index.index)
 
     tm.assert_frame_equal(joined, expected)
+
+
+def test_join_left_sequence_non_unique_index(left, right, right_non_unique):
+    # left join sequence of dataframes with non-unique indices (issue #19607)
+    joined = left.join([right_non_unique], how='left')
```

Comment:
Could you also build the expected output and `tm.assert_frame_equal` against that?

Reply:
That should definitely be possible